doc/hooks/patch-activation-hooks.section.md

```diff
@@ -0,0 +1,51 @@
+
+# `patchActivationBash`, `patchActivationCsh`, `patchActivationFish`, `patchActivationPosix` {#sec-patchActivationHooks}
+
+These hooks provide utilities to patch
+shell scripts meant to be sourced (e.g. the "activate" scripts).
+
+It inserts two sections of codes into the beginning and the end of the script:
+the former inserts a `PATH` prefix, while the latter clears it up.
+This way, the original commands will have access to the executables needed,
+but how `PATH` variable is changed by the script remains the same after patching.
+
+A stack is used to record the inserted `PATH` prefixes when patched multiple times,
+so that a patched activation script can correctly source another patched script.
```

Comment:
Not sure if the implementation detail is all that important, and while I don't oppose including it, I think it can be shorter. Don't have a concrete suggestion though, sorry. Maybe something like "For each supported shell, creates a context with a `PATH` prefix for the given script, which is exited when the script ends, and allows nested contexts" (but this may be thought in a too Pythonic way).